### ISSUE TITLE
Implement separate user/password variables for rest-server backend

### ISSUE THREAD

**Issue by TheForcer** [cupcakearmy/autorestic#48]
Hi,

I am currently using restic in combination with a shell script, mostly with local & rest-server backens, on my servers. So far, autorestic seems perfect to replace my shell-scripts with a much nicer high-level interface.
A small improvement I would love to see is separate credentials variables in the config for the rest-server backend, as I use it with the integrated HTTP auth. Maybe something along those lines:

```
backends:
  name-of-backend:
    type: rest
    path: https://restserver.example.com/johndoe
    user: johndoe
    password: supersecurepw
```

Of course I could still put those in the URL, but https://johndoe:supersecurepw@restserver.example.com/johndoe is very unwieldy and prone to typos. Whaat do you think? :)

**Comment by cupcakearmy** [comment]
It's an interesting suggestion. I could implement it, as optional.
However I would suggest a submenu, don't want to pollute the global config.

```yaml
backends:
  name-of-backend:
    type: rest
    path: restserver.example.com/johndoe
    rest:
      user: johndoe
      password: supersecurepw
```

What do you think? If no user and password are in the `rest` field the `path` would be taken unchanged to ensure back compatibility.

**Comment by TheForcer** [comment]
Sounds good! Backwards compatibility definitely makes sense, since not everyone is using the auth functionality. Your suggestion should work just fine.

**Comment by cupcakearmy** [comment]
@TheForcer should be fixed in v1.0.4
You can run `autorestic upgrade` to do that automatically :)

**Comment by TheForcer** [comment]
Wow, awesome! Thank you for implementing this already :)
